builtin/game/misc.lua

```diff
@@ -4,74 +4,42 @@
 -- Misc. API functions
 --
 
-local timers = {}
-local mintime
-local function update_timers(delay)
-	mintime = false
-	local sub = 0
-	for index = 1, #timers do
-		index = index - sub
-		local timer = timers[index]
-		timer.time = timer.time - delay
-		if timer.time <= 0 then
-			core.set_last_run_mod(timer.mod_origin)
-			timer.func(unpack(timer.args or {}))
-			table.remove(timers, index)
-			sub = sub + 1
-		elseif mintime then
-			mintime = math.min(mintime, timer.time)
-		else
-			mintime = timer.time
-		end
-	end
-end
+local _jobs = {}
+local _time = 0.0
+local _last = 0.0
 
-local timers_to_add
-local function add_timers()
-	for _, timer in ipairs(timers_to_add) do
-		table.insert(timers, timer)
+core.register_globalstep(function(dtime)
+	local _new = core.get_us_time() / 1000000
+	if _new > _last then
+		_time = _time + (_new - _last)
+	else
+		-- rollover
+		_time = _time + _new
```

Comment:
This isn't quite correct, the real dtime will be a bit more (`UINT32_MAX / 1000000 - last` more, if my math's right).
I suppose it doesn't matter much if this only happens rarely, but I'd fall back to `dtime` in this case (assuming `dtime` doesn't go haywire on overflow).